pkg/network/usm/ebpf_ssl.go

```diff
@@ -694,3 +714,57 @@ func deleteDeadPidsInMap(manager *manager.Manager, mapName string, alivePIDs map
 
 	return nil
 }
+
+// deleteDeadPidsInSSLCtxMap finds a map by name and deletes dead processes.
+func deleteDeadPidsInSSLCtxMap(manager *manager.Manager, alivePIDs map[uint32]struct{}) error {
+	sslCtxByPIDTGIDMapObj, _, err := manager.GetMap(sslCtxByPIDTGIDMap)
+	if err != nil {
+		return fmt.Errorf("dead process ssl cleaner failed to get map: %q error: %w", sslCtxByPIDTGIDMap, err)
+	}
+	sslSockByCtxMapObj, _, err := manager.GetMap(sslSockByCtxMap)
+	if err != nil {
+		return fmt.Errorf("dead process ssl cleaner failed to get map: %q error: %w", sslSockByCtxMap, err)
+	}
+	sslCtxByTupleMapObj, _, err := manager.GetMap(sslCtxByTupleMap)
+	if err != nil {
+		return fmt.Errorf("dead process ssl cleaner failed to get map: %q error: %w", sslCtxByTupleMap, err)
+	}
+
+	var pidKeysToDelete []uint64
+	var sockKeysToDelete []uintptr
+	var tupleKeysToDelete []http.ConnTuple
+
+	var key uint64
+	var value uintptr
+	iter := sslCtxByPIDTGIDMapObj.Iterate()
+
+	for iter.Next(unsafe.Pointer(&key), unsafe.Pointer(&value)) {
+		pid := uint32(key >> 32)
+		if _, exists := alivePIDs[pid]; !exists {
+			pidKeysToDelete = append(pidKeysToDelete, key)
+
+			sslCtxKey := value
+			sockKeysToDelete = append(sockKeysToDelete, sslCtxKey)
+
+			var sock http.SslSock
+			if err := sslSockByCtxMapObj.Lookup(unsafe.Pointer(&sslCtxKey), unsafe.Pointer(&sock)); err == nil {
+				tupleKeysToDelete = append(tupleKeysToDelete, sock.Tup)
+			}
```

Comment:
another thing to consider, add the pid or tgid_pid to the key of that map, it will allow us easier deletion